Shared/ThresholdKeyView.swift

```diff
@@ -129,6 +131,21 @@ struct ThresholdKeyView: View {
                                     showAlert = true
                                     tkeyReconstructed = true
                                     resetAccount = false
+                                    let shareIndexes = try threshold_key.get_shares_indexes()
+                                    
+                                    // let's get the device share index
+                                    var deviceShareIndex = ""
+                                    // "1" is for the social login share
+                                    if shareIndexes[0] == "1" {
+                                        deviceShareIndex = shareIndexes[1]
+                                    } else {
+                                        deviceShareIndex = shareIndexes[0]
+                                    }
+                                    
+                                    let share = try! threshold_key.output_share(shareIndex: deviceShareIndex, shareType: nil)
+                                    let saveId = fetchKey + ":0"
+                                    //save the device share
+                                    try! KeychainInterface.save(item: share, key: saveId)
```

Comment:
The `try!` will crash if it fails
```suggestion
                                    let share = try threshold_key.output_share(shareIndex: deviceShareIndex, shareType: nil)
                                    let saveId = fetchKey + ":0"
                                    //save the device share
                                    try KeychainInterface.save(item: share, key: saveId)
```